crates/node/src/service/rpc/middleware.rs

```diff
@@ -267,18 +267,30 @@ async fn add_rpc_version_to_method(req: &mut hyper::Request<Body>) -> Result<(),
     let version = RpcVersion::from_request_path(&path)?;
 
     let whole_body = hyper::body::to_bytes(req.body_mut()).await?;
-    let mut json: Value = serde_json::from_slice(&whole_body)?;
+    let json: Value = serde_json::from_slice(&whole_body)?;
 
-    if let Some(method) = json.get_mut("method").as_deref().and_then(Value::as_str) {
-        let new_method = format!("starknet_{}_{}", version.name(), method.strip_prefix("starknet_").unwrap_or(method));
-
-        json["method"] = Value::String(new_method);
+    // in case of batched requests, the request is an array of JSON-RPC requests
+    let mut batched_request = false;
+    let mut items = if let Value::Array(items) = json {
+        batched_request = true;
+        items
     } else {
-        return Err(VersionMiddlewareError::InvalidRequestFormat);
+        vec![json]
+    };
```

Comment:
Could we do an helper function for batched_request like:

```rust
fn is_batched_request(json: &Value) -> Option<&Vec<Value>> {
    match json {
        Value::Array(items) => Some(items),
        _ => None,
    }
}
```

Reply:
Not mandatory but it would be cleaner

Reply:
Hmm, although making `is_batched_request` return a optional vector looks non intuitive. I can maybe change it to `get_items_from_request` but then I would need to return a tuple - (items, is_batched_request)